controller/authorize.go

```diff
@@ -0,0 +1,93 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/login"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/fabric8-services/fabric8-auth/token"
+	"github.com/goadesign/goa"
+	errs "github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+// AuthorizeController implements the authorize resource.
+type AuthorizeController struct {
+	*goa.Controller
+	Auth          login.KeycloakOAuthService
+	TokenManager  token.Manager
+	Configuration LoginConfiguration
+}
+
+func NewAuthorizeController(service *goa.Service, auth *login.KeycloakOAuthProvider, tokenManager token.Manager, configuration LoginConfiguration) *AuthorizeController {
+	return &AuthorizeController{Controller: service.NewController("AuthorizeController"), Auth: auth, TokenManager: tokenManager, Configuration: configuration}
+}
+
+// Authorize runs the authorize action.
+func (c *AuthorizeController) Authorize(ctx *app.AuthorizeAuthorizeContext) error {
+
+	if ctx.State == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("state", "nil").Expected("state"))
+	}
+	var scope []string
+
+	if ctx.Code == nil {
+		if ctx.ClientID == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("client_id", "nil").Expected("service account id"))
+		}
+		if ctx.RedirectURI == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("redirect_uri", "nil").Expected("redirect uri"))
+		}
+		if ctx.ResponseType == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("response_type", "nil").Expected("response type"))
+		}
+
+		if ctx.Scope == nil {
+			scope = []string{"user:email"}
+		} else {
+			scope = []string{*ctx.Scope}
+		}
+
+		_, found := c.Configuration.GetServiceAccounts()[*ctx.ClientID]
```

Comment:
Did you mean https://github.com/fabric8-services/fabric8-auth/blob/master/configuration/conf-files/service-account-secrets.conf this file?

Reply:
@alexeykazakov @sbose78 
since you want types, I think It would need everything to be refactored/renamed, sort of like
service-account-secrets.conf --> account-secrets.conf
type ServiceAccount --> type Account
and so on. 
I did a quick search of occurences of "serviceAccount". Quite a bit of renaming. I just want to make sure that I am going the right way.  ...?

Reply:
We can do renaming in a separate PR.